### ISSUE TITLE
Problem in reproducing GenEval score (0.75) with released Omni-Video checkpoint

### ISSUE THREAD

**Issue by leduy99** [SAIS-FUXI/Omni-Video#11]
Hi, thanks a lot for releasing Omni-Video and the training code!

I’m trying to reproduce the **GenEval** results reported in the paper (Table 2, overall **0.75** for T2I).  
With the released checkpoint and current code, I only obtain **0.522** overall. I’d like to confirm whether my setup matches yours.

---

### 1. Setup

- Model: Omni-Video 1.3B T2I (Wan 2.1 backbone) from this repo.
- Code: latest `main` branch (`tools/inference/generate.py`, `configs/foster/omnivideo_mixed_task_1_3B.yaml`).
- GenEval: `djghosh13/geneval`, default pipeline (`evaluate_images.py` + `summary_scores.py`).
- Resolution for GenEval: **512×512**, 1 image per prompt (553 prompts).

---

### 2. Generation settings for GenEval

```python
# Task
task = "t2i"
frame_num = 1

# Sampling
sample_solver = "unipc"
sample_steps = 50
sample_guide_scale = 5.0
sample_shift = 5.0

# Conditioning / adapters
condition_mode = "full"
use_visual_as_input = False
use_visual_context_adapter = True
max_context_len = 2560

# AR (rethinking)
ar_model_num_video_frames = 8
ar_conv_mode = "llama_3"
ar_query = "<video>\n Describe this video and its style in a very detailed manner."
# AR output is also used for task detection with tokens 128002/128003.

# Classifier-free guidance
classifier_free_ratio = 0.0
unconditioned_context_length = 2560

# Negative prompt (from wan_shared_cfg.sample_neg_prompt)
sample_neg_prompt = (
  "色调艳丽，过曝，静态，细节模糊不清，字幕，风格，作品，画作，画面，静止，整体发灰，"
  "最差质量，低质量，JPEG压缩残留，丑陋的，残缺的，多余的手指，画得不好的手部，画得不好的脸部，"
  "畸形的，毁容的，形态畸形的肢体，手指融合，静止不动的画面，杂乱的背景，三条腿，背景人很多，倒着走"
)
````

---

### 3. GenEval results

Prompt-level accuracy per task:

* **position:** 14.25% (57 / 400)
* **color_attr:** 20.00% (80 / 400)
* **counting:** 51.88% (166 / 320)
* **single_object:** 94.38% (302 / 320)
* **colors:** 73.67% (277 / 376)
* **two_object:** 59.09% (234 / 396)

**Overall GenEval score (avg over tasks): 0.52210**

This is much closer to the early ~0.49 result mentioned in the paper than to the final **0.75**.

---

### 4. Questions

1. **Checkpoint**

   * Is the released Omni-Video checkpoint the same one used for the **0.75 GenEval** result, or was that from a later internal checkpoint / additional fine-tuning?

2. **Rethinking / AR usage**
   * Originally, I didn't use this mode and achieved the score of **0.49**, the same as the score reported in the early versions of the paper. After using this, score jumped to **0.52**. So:
   * For GenEval, did you enable AR “rethinking” (and visual context adapter), or run a **pure T2I** pipeline with text-only conditioning?

3. **Task detection**

   * For GenEval, did you rely on AR-based task detection (tokens 128002/128003), or did you **force `task="t2i"`** for all prompts?

4. **Sampling hyperparameters**

   * What exact values did you use for `sample_shift`, `num_inference_steps`, guidance scale, and resolution in the official GenEval evaluation?

5. **Negative prompt**

   * Is the `wan_shared_cfg.sample_neg_prompt` string above the exact “universal negative prompt” you used for GenEval?

If possible, could you share the exact config / command line (or a small script) used to obtain the 0.75 GenEval score? That would be extremely helpful to align my reproduction with your setup.

Thank you again for your work and for any clarification!
